### DISCUSSION TITLE
Dashboard beta ending - next steps

### DISCUSSION THREAD

**Discussion by ahpook** [community/OSPO#89] [Dashboard]
> TL;DR - we will be turning down the Organization Metrics Dashboard UI by the end of September; instead, we'll focus on providing API access to the underlying metrics as well as reference implementations on third-party dashboarding packages.

Hi all - over the past few months we've gotten great feedback from beta users of the Organization Metrics Dashboard. We iterated on the design to make it easier to use, added useful metrics that were initially missing, and sped up the response times for large repos. Unfortunately, the internal data platform we used to store the time-series data is deprecated due to shifting company priorities, and it'd be a big engineering challenge to re-create that capability. So we've made the difficult but necessary call to exit the beta phase by removing the dashboard rather than moving forward to GA. This will be effective by Sept 2023, when the data platform is decommissioned.

As a replacement, we intend to provide:

* API access to the metrics data, if there is anything currently missing
* Reference implementations on third-party dashboard/data viz solutions
* Documentation/HOWTOs on adding the data to your own dashboard packages

The leading contender for a reference implementation is [Cauldron.io](https://cauldron.io), which is a SaaS implementation of the open source [Grimoire Lab project](https://chaoss.github.io/grimoirelab/), closely tied to the [CHAOSS community metrics](https://chaoss.community) we used as guidance for our own dashboard. We're also investigating more general-purpose dashboarding packages like PowerBI and Grafana.

I'd love to hear what would be most useful to you, in terms of building visualizations for community health metrics based on GitHub data. If you've got your own dashboards today, what are you using? What would you like see from the API that it doesn't currently provide? And how can we best replace the Org Metrics Dashboard use-cases that matter most to you? Please let us know in this discussion!

**Comment by terrchen** [comment]
I just found this discussion (followed a rabbit hole that started at https://github.com/github/github-ospo) and I thought I would share what I've been working on, that might be of an interest for others.  I recently took my DevBoard out of stealth mode a few weeks ago and you can see it in action for the electron org at https://devboard.gitsense.com/electron.   DevBoard (the dashboard system) will be open sourced in the coming months and you can learn more about the dashboard widgets at https://devboard.gitsense.com/electron?board=gitsense_examples.intro_to_widgets  Note, the repo mentioned in the intro page hasn't been pushed to GitHub yet.

I don't have much to add regarding API access but I would like to open a discussion regarding making it easier to integrate third-party dashboard systems with GitHub.  Right now, I have a Chrome extension that will allow you to browse DevBoard inside GitHub, but it's kind of janky.  It's okay, but it's not as smooth as I would like and I was wondering if this can be discussed further, as I do have a suggestion.

**Reply by ahpook** [reply]
@terrchen This is very cool, thanks for sharing!

Re: integrating dashboard systems, do you mean like embedding js apps in the browser experience for repo views? There is a project called Blocks that is intended to solve for that problem:  https://blocks.githubnext.com/

**Reply by terrchen** [reply]
@ahpook I was thinking more about making it easier to add a new tab component.  You can see how the extension works right now at https://gitsense.com/DevBoard-Chrome-Extension.m4v  Right now the extension piggy backs off "insights" when in a repo and "people" when in a org.

**Comment by binford2k** [comment]
PowerBI is what Perforce uses internally, although cauldron or grafana would probably also work for us.
